### PR TITLE
Use global store for connection status

### LINKED ISSUE #913: Use global store for connection status provider

### PR DESCRIPTION
**Pull request by mikkio-j**
Issue: #913

### CODE REVIEW

File: src/providers/connectionStatus.tsx [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-  const [nodeConnectionStatus, setNodeConnection] = useState<ConnectionStatus>('connecting')
-  const [isUserConnectedToInternet, setIsUserConnectedToInternet] = useState(true)
+export const useConnectionStatusStore = create<ConnectionStatusValue>((set) => ({
+  isUserConnectedToInternet: true,
+  nodeConnectionStatus: 'connecting',
+  setNodeConnection: (connection) => {
+    set((state) => ({
+      ...state,
+      nodeConnectionStatus: connection,
+    }))
```
Comment (drillprop) [inline src/providers/connectionStatus.tsx]:
```suggestion
    set((state) => {
      state.nodeConnectionStatus = connection
    })
```

File: src/providers/connectionStatus.tsx [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+  },
+  setInternetConnection: (value) => {
+    set((state) => ({
+      ...state,
+      isUserConnectedToInternet: value,
+    }))
```
Comment (drillprop) [inline src/providers/connectionStatus.tsx]:
```suggestion
    set((state) => {
      state.isUserConnectedToInternet = value
    })
```

**Review by drillprop — COMMENTED** [review]
Works good, but you forgot that we are using `immer`. Added two suggestions

File: src/providers/connectionStatus.tsx [hunk 3]
```diff
@@ -1,23 +1,37 @@
...
 
-const ConnectionStatusContext = React.createContext<undefined | ConnectionStatusValue>(undefined)
-ConnectionStatusContext.displayName = 'ConnectionStatusContext'
-
 type ConnectionStatusValue = {
-  setNodeConnection: (connection: ConnectionStatus) => void
   nodeConnectionStatus: ConnectionStatus
   isUserConnectedToInternet: boolean
+  setNodeConnection: (connection: ConnectionStatus) => void
+  setInternetConnection: (value: boolean) => void
```
Comment (drillprop) [inline src/providers/connectionStatus.tsx]:
```suggestion
  setInternetConnection: (isConnected: boolean) => void
```
last small thing - I think we should be more semantic here, let's just rename it

File: src/providers/connectionStatus.tsx [hunk 4]
```diff
@@ -1,23 +1,37 @@
...
+  isUserConnectedToInternet: true,
+  nodeConnectionStatus: 'connecting',
+  setNodeConnection: (connection) => {
+    set((state) => {
+      state.nodeConnectionStatus = connection
+    })
+  },
+  setInternetConnection: (value) => {
+    set((state) => {
+      state.isUserConnectedToInternet = value
```
Comment (drillprop) [inline src/providers/connectionStatus.tsx]:
```suggestion
  setInternetConnection: (isConnected) => {
    set((state) => {
      state.isUserConnectedToInternet = isConnected
```

**Review by drillprop — APPROVED** [review]

File: src/providers/connectionStatus.tsx [hunk 5]
```diff
@@ -1,23 +1,37 @@
...
+  setNodeConnection: (connection: ConnectionStatus) => void
+  setInternetConnection: (isConnected: boolean) => void
 }
 
 const SNACKBAR_TIMEOUT = 15000
 
-export const ConnectionStatusProvider: React.FC = ({ children }) => {
-  const [nodeConnectionStatus, setNodeConnection] = useState<ConnectionStatus>('connecting')
-  const [isUserConnectedToInternet, setIsUserConnectedToInternet] = useState(true)
+export const useConnectionStatusStore = create<ConnectionStatusValue>((set) => ({
```
Comment (kdembler) [inline src/providers/connectionStatus.tsx]: This is using the default `zustand` `create`, not from out store utilities. We should use the common helper. This one also doesn't have `immer` so my guess is it's not fully working

**Review by kdembler — CHANGES_REQUESTED** [review]

File: src/views/studio/SignInView/SignInView.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ import {
 export const SignInView = () => {
   const navigate = useNavigate()
   const { activeChannelId, setActiveUser, memberships } = useUser()
-  const { nodeConnectionStatus } = useConnectionStatus()
+  const nodeConnectionStatus = useConnectionStatusStore((state) => state.nodeConnectionStatus)
```
Comment (drillprop) [inline src/views/studio/SignInView/SignInView.tsx:25]: Previously if there was no internet connection all the tiles and sign-in button were disabled. Right now if you disconnect from wifi, you'll still be able to click those.

**Review by kdembler — APPROVED** [review]

**Pull request merged**